packages/input-gateway/src/sendQueue.ts

```diff
@@ -0,0 +1,66 @@
+type AsyncFunction = () => Promise<void>
+
+interface JobDescription {
+	className?: string
+	resolve: () => void
+	reject: (error?: any) => void
+	fn: AsyncFunction
+}
+
+type AddOptions = {
+	className?: string
+}
+
+type Options = {
+	autoStart: boolean
+}
+
+export class SendQueue {
+	#queue: JobDescription[] = []
+	#autoStart: boolean
+	#paused = true
+
+	constructor(opts?: Options) {
+		this.#autoStart = opts?.autoStart ?? true
+	}
+
+	async add(fn: AsyncFunction, options?: AddOptions): Promise<void> {
+		return new Promise((resolve, reject) => {
+			this.#queue.push({
+				className: options?.className,
+				fn,
+				resolve,
+				reject,
+			})
+			if (this.#autoStart && this.#paused) this.start()
+		})
+	}
+
+	clear(): void {
+		this.#queue.length = 0
+	}
+
+	remove(className?: string): void {
+		this.#queue = this.#queue.filter((job) => job.className !== className)
+	}
+
+	start(): void {
```

Comment:
should this have a guard of `if (!this.#paused) return`?

That combined with L35, if autoStart is set to false, then `start()` will have to be called a lot to ensure the queue runs